pkg/registry/batch/job/strategy.go

```diff
@@ -213,6 +210,14 @@ func (jobStrategy) WarningsOnCreate(ctx context.Context, obj runtime.Object) []s
 	return warnings
 }
 
+// generateSelectorIfNeeded checks the job's manual selector flag and generates selector labels if the flag is true.
+func generateSelectorIfNeeded(obj *batch.Job) {
+	// TODO: move UID generation earlier and do this in defaulting logic?
+	if obj.Spec.ManualSelector == nil || *obj.Spec.ManualSelector == false {
```

Comment:
Can you verify if you have tests that are actually hitting the case of true?

Reply:
Test cases in the `TestJobStrategy_PrepareForCreate` are hitting this if clause.